src/party_menu.c

```diff
@@ -5753,10 +5767,25 @@ static void PartyMenuTryEvolution(u8 taskId)
     }
     else
     {
-        if (gPartyMenu.menuType == PARTY_MENU_TYPE_FIELD && CheckBagHasItem(gSpecialVar_ItemId, 1))
-            gTasks[taskId].func = Task_ReturnToChooseMonAfterText;
+        targetSpecies = GetEvolutionTargetSpecies(mon, EVO_MODE_CANT_STOP, ITEM_NONE, NULL);
+
+        if (targetSpecies != SPECIES_NONE)
+        {
+            FreePartyPointers();
+            if (ItemId_GetFieldFunc(gSpecialVar_ItemId) == ItemUseOutOfBattle_RareCandy && gPartyMenu.menuType == PARTY_MENU_TYPE_FIELD && CheckBagHasItem(gSpecialVar_ItemId, 1))
+                gCB2_AfterEvolution = CB2_ReturnToPartyMenuUsingRareCandy;
+            else
+                gCB2_AfterEvolution = gPartyMenu.exitCallback;
+            BeginEvolutionScene(mon, targetSpecies, FALSE, gPartyMenu.slotId);
+            DestroyTask(taskId);
+        }
         else
-            gTasks[taskId].func = Task_ClosePartyMenuAfterText;
+        {
+            if (gPartyMenu.menuType == PARTY_MENU_TYPE_FIELD && CheckBagHasItem(gSpecialVar_ItemId, 1))
+                gTasks[taskId].func = Task_ReturnToChooseMonAfterText;
+            else
+                gTasks[taskId].func = Task_ClosePartyMenuAfterText;
+        }
```

Comment:
Could you explain why this change was needed?

Reply:
Since EVO_MODE_CANT_STOP is the same as EVO_MODE_NORMAL but you cant stop the evolution, I had to edit every place that called BeginEvolutionScene or EvolutionScene from EVO_MODE_NORMAL and add a condition that checks for EVO_MODE_CANT_STOP as well.

Reply:
Could see if you can rewrite it? So you avoid duplicating code:

```
if (check) 
    BeginEvolutionScene(mon, targetSpecies, TRUE, gPartyMenu.slotId);
else
    BeginEvolutionScene(mon, targetSpecies, FALSE, gPartyMenu.slotId);
```

Reply:
The code is somewhat confusing right now because of the duplication and I think there is a way to do it with minimal changes.